src/test/parse-fail/obsolete-proc.rs

```diff
@@ -10,8 +10,8 @@
 
 // Test that we generate obsolete syntax errors around usages of `proc`.
 
-fn foo(p: proc()) { } //~ ERROR obsolete syntax: the `proc` type
+fn foo(p: proc()) { } //~ ERROR: the `proc` type isn't used for the moment
```

Comment:
What I'm saying is that this error message cannot happen an this test needs to be updated if it wants to get past the bots. The `ObsoleteSyntax` changes you made above are never actually called (`ProcExpr` is never referenced and `ProcType` is only called in a dead function).

Could you:

* Delete `ProcExpr` and `ProcType`
* Delete `parse_proc_type`
* Update the error messages in this test